element_deeplabcut/dlc.py

```diff
@@ -142,154 +127,123 @@ class File(dj.Part):
-            :param displayiters: int, display iterations
-            :param saveiters: int, save X iterations
-            :param maxiters: int, maximum iterations
-        :param allow_growth: bool, default false. True will not pre-allocate gpu memory
-        :param gputouse: int, optional. number of your GPU (see number in nvidia-smi).
-        :param autotune: bool TensorFlow property, faster if 'false'. Default false
-        :param keepdeconvweights: bool, default: true. restore weights from snapshot
-                                  If changing bodyparts, set to true
-        """
+    def make(self, key):
+        import inspect
         from deeplabcut import train_network
-        from .readers.dlc_reader import PoseEstimation
+        from deeplabcut.utils.auxiliaryfunctions import GetModelFolder
 
-        train_key = (TrainingTask & f'training_id={training_id}').fetch1('KEY')
-        project_path = find_full_path(get_dlc_root_data_dir(),
-                                      get_session_directory(train_key))
-        model = PoseEstimation(project_path)
-        params = (ConfigParamSet & train_key).fetch1()
-        if params['train_fraction'] not in model.yml['TrainingFraction']:
-            # Config lists fractions used. If user provided val not exist, raise Err
-            # TODO: move duplicated code to reader
-            raise FileNotFoundError('Train Fraction {train_fraction}'.format(**params)
-                                    + ' was not found in the config file\n'
-                                    + model.yml_path)
-        train_fract_idx = model.yml['TrainingFraction'].index(params['train_fraction'])
+        training_id, project_path, model_prefix = (TrainingTask & key).fetch1(
+            'training_id', 'project_path', 'model_prefix')
+
+        project_path = find_full_path(get_dlc_root_data_dir(), project_path)
+
+        # ---- Build and save DLC configuration (yaml) file ----
+        dlc_config = (ModelTrainingParamSet & key).fetch1('params')
+        dlc_config['project_path'] = dlc_project_path.as_posix()
+        dlc_config['modelprefix'] = model_prefix
+
+        video_filepaths = [find_full_path(get_dlc_root_data_dir(), fp).as_posix()
+                           for fp in (VideoRecording.File & key).fetch('file_path')]
+        dlc_config['video_sets'] = video_filepaths
+
+        # ---- Write DLC and basefolder yaml (config) files ----
+
+        # Write dlc config file to base (data) folder
+        # This is important for parsing the DLC in datajoint imaging
+        output_dir.mkdir(exist_ok=True)
+        dlc_cfg_filepath = project_path / 'config.yaml'
+        with open(dlc_cfg_filepath, 'w') as f:
+            yaml.dump(dlc_config, f)
+
+        # ---- Trigger DLC model training job ----
+        train_network_input_args = list(inspect.signature(train_network).parameters)
+        train_network_kwargs = {k: v for k, v in dlc_config.items() if k in train_network_input_args}

```

Comment:
It looks like these lines 
1. Ask what are the arguments that DLC's `train_network` takes
2. Of those, keep only applicable config items
Is that right?